File: API/Placeful.Api/Models/Location.cs

using Placeful.Models;

namespace Placeful.Api.Models;
public class Location
{
    public Guid Id { get; set; }

    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public IEnumerable<Memory> Memories { get; set; } = new List<Memory>();
}

File: Frontend/placeful/lib/core/domain/models/location.dart
import 'package:json_annotation/json_annotation.dart';
import 'memory.dart';

part 'location.g.dart';

@JsonSerializable(explicitToJson: true)
class Location {
  final String id;
  final double latitude;
  final double longitude;
  final List<Memory> memories;

  Location({
    required this.id,
    required this.latitude,
    required this.longitude,
    this.memories = const [],
  });

  factory Location.fromJson(Map<String, dynamic> json) =>
      _$LocationFromJson(json);

  Map<String, dynamic> toJson() => _$LocationToJson(this);
}
